src/replication.c

```diff
@@ -1570,8 +1570,9 @@ void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData,
             serverLog(LL_WARNING,"Diskless rdb transfer, last replica dropped, killing fork child.");
             killRDBChild();
         }
-        /*  Remove the pipe read handler if at least one write handler was set. */
-        if (server.rdb_pipe_numconns_writing || stillAlive == 0) {
+        /* Remove the pipe read handler if at least one write handler was set,
+         * or if we are restarting a BGSAVE for other replicas. */
+        if (server.rdb_pipe_numconns_writing || (stillAlive == 0 && !hasActiveChildProcess())) {
```

Comment:
how did you find it?
is this a result of the new change in killRDBChild?
i'm starting to think maybe we should revert that change and find another way to avoid a corrupt dirty counter after flushall.

maybe a nicer solution is to re-order these `if` blocks and have the call to killRDBChild done below the one for aeDeleteFileEvent?

Reply:
failed CI: https://github.com/redis/redis/actions/runs/9848662904/job/27191018142
through the logs, the master started BGSAVE, but the replica didn't receive any message from master for almost 10 seconds.